src/Caching/Cache.php

```diff
@@ -98,6 +98,43 @@ public function load($key, $fallback = NULL)
 
 
 	/**
+	 * Reads multiple items from the cache
+	 * @param array
+	 * @param callable
+	 * @return array
+	 */
+	public function bulkLoad(array $keys, $fallback = NULL)
+	{
+		foreach ($keys as $key) {
+			if (!is_scalar($key)) {
+				throw new Nette\InvalidArgumentException('Only scalar keys are allowed in a bulkLoad method.');
+			}
+		}
+		$storageKeys = array_map([$this, 'generateKey'], $keys);
+		$storage = $this->getStorage();
+		if ($storage instanceof IBulkReadStorage) {
+			$cacheData = $storage->bulkRead($storageKeys);
+		} else {
+			$cacheData = array_combine($storageKeys, array_map([$storage, 'read'], $storageKeys));
```

Comment:
Rewrite this to 

~~~php
$result = array_combine($keys, array_map([$storage, 'read'], $storageKeys));

foreach ($result as $key => $value) {
    if ($value === NULL) {
        $result[$key] = $this->save($key, ...);
    }
}
~~~